wsagent/che-core-api-git/src/test/java/org/eclipse/che/git/impl/LogTest.java

```diff
@@ -114,4 +120,206 @@ public void testLogWithFileFilter(GitConnectionFactory connectionFactory) throws
+
+        assertEquals(1, secondBucketOfCommits.size());
+
+        assertEquals(secondBucketOfCommits.get(0).getMessage(), "add 1.txt file");
+        assertEquals(secondBucketOfCommits.get(0).getBranches().get(0).getName(), "refs/heads/master");
+        assertEquals(secondBucketOfCommits.get(0).getDiffCommitFile().get(0).getOldPath(), "/dev/null");
+        assertEquals(secondBucketOfCommits.get(0).getDiffCommitFile().get(0).getNewPath(), "1.txt");
+        assertEquals(secondBucketOfCommits.get(0).getDiffCommitFile().get(0).getChangeType(), "ADD");
+    }
+
+    @Test(dataProvider = "GitConnectionFactory", dataProviderClass = GitConnectionFactoryProvider.class)
+    public void testLogMaxCount(GitConnectionFactory connectionFactory) throws GitException, IOException {
+        //given
+        GitConnection connection = connectToInitializedGitRepository(connectionFactory, repository);
+        addFile(connection, "1.txt", "someChanges");
+        connection.add(newDto(AddRequest.class));
+        connection.commit(newDto(CommitRequest.class).withMessage("add 1.txt file"));
+
+        addFile(connection, "2.txt", "newChanges");
+        connection.add(newDto(AddRequest.class));
+        connection.commit(newDto(CommitRequest.class).withMessage("add 2.txt file"));
+
+        addFile(connection, "3.txt", "otherChanges");
+        connection.add(newDto(AddRequest.class));
+        connection.commit(newDto(CommitRequest.class).withMessage("add 3.txt file"));
+
+        addFile(connection, "4.txt", "someChanges");
+        connection.add(newDto(AddRequest.class));
+        connection.commit(newDto(CommitRequest.class).withMessage("add 4.txt file"));
+
+        //when
+        List<Revision> allCommits =
+            connection.log(newDto(LogRequest.class)).getCommits();
+        List<Revision> firstBucketOfCommits =
+            connection.log(newDto(LogRequest.class).withMaxCount(4)).getCommits();
+        List<Revision> secondBucketOfCommits =
+            connection.log(newDto(LogRequest.class).withMaxCount(2)).getCommits();
+
+        //then
+        assertEquals(4, allCommits.size());
+
+        assertEquals(4, firstBucketOfCommits.size());
+        assertEquals(firstBucketOfCommits.get(0).getMessage(), "add 4.txt file");
+        assertEquals(firstBucketOfCommits.get(1).getMessage(), "add 3.txt file");
+        assertEquals(firstBucketOfCommits.get(2).getMessage(), "add 2.txt file");
+        assertEquals(firstBucketOfCommits.get(3).getMessage(), "add 1.txt file");
+
+        assertEquals(2, secondBucketOfCommits.size());
+        assertEquals(secondBucketOfCommits.get(0).getMessage(), "add 4.txt file");
+        assertEquals(secondBucketOfCommits.get(1).getMessage(), "add 3.txt file");
+    }
+
+    @Test(dataProvider = "GitConnectionFactory", dataProviderClass = GitConnectionFactoryProvider.class)
+    public void testLogWithSkipAndMaxCount(GitConnectionFactory connectionFactory) throws GitException, IOException {
```

Comment:
Because i added three tests:
1) testLogSkip -  to check log with skip option
2) testLogMaxCount - to check log with maxCount option
3)testLogWithSkipAndMaxCount -  to check log with skip and maxCount together

In addition i added more tests